test/parallel/test-module-children.js

```diff
@@ -0,0 +1,12 @@
+// Flags: --no-deprecation
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const dir = common.fixturesDir + '/GH-7131';
```

Comment:
> @refack to be clear, you're saying that path.join() is wrong because it'll resolve to a\\b on Windows, and require always takes a POSIX style path (a/b)?

Not wrong per se, just less right.